src/core/ext/transport/chttp2/transport/hpack_parser.cc

```diff
@@ -764,33 +766,44 @@ class HPackParser::String {
   // decoded byte.
   template <typename Out>
   static bool ParseHuff(Input* input, uint32_t length, Out output) {
-    int16_t state = 0;
-    // Parse one half byte... we leverage some lookup tables to keep the logic
-    // here really simple.
-    auto nibble = [&output, &state](uint8_t nibble) {
-      int16_t emit = emit_sub_tbl[16 * emit_tbl[state] + nibble];
-      int16_t next = next_sub_tbl[16 * next_tbl[state] + nibble];
-      if (emit != -1) {
-        if (emit >= 0 && emit < 256) {
-          output(static_cast<uint8_t>(emit));
-        } else {
-          assert(emit == 256);
+    if (IsNewHpackHuffmanDecoderEnabled()) {
+      // If there's insufficient bytes remaining, return now.
```

Comment:
Looks like this this if -` if (input->remaining() < length)... ` is the same in the bottom else part. We should be able to bring it out

Reply:
done